lib/core/macros.nim

```diff
@@ -667,8 +667,18 @@ template expectRoutine(node: NimNode): stmt =
   expectKind(node, RoutineNodes)
 
 proc name*(someProc: NimNode): NimNode {.compileTime.} =
+  ## Returns name node of the proc. Note: if result is nnkPostfix
+  ## the name is not unpacked. Use `unpackedName` to obtain the "pure" name.
   someProc.expectRoutine
   result = someProc[0]
+
+proc unpackedName*(someProc: NimNode): NimNode {.compileTime.} =
```

Comment:
You probably mean `unpackPostfix`, but that one expects only `nnkPostfix` nodes.